File: vehicle-service-quarkus-app/src/main/java/com/redbend/model/Ecu.java
package com.redbend.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.List;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name="ECU")
@JsonIgnoreProperties({"vehicle"})
public class Ecu {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String manufacturer;
    private String model;

    @OneToMany(mappedBy = "ecu")
    private List<Software> softwareList;

    @ManyToOne
    private Vehicle vehicle;
}

File: update-service/src/main/kotlin/com/demo/server/ecu/Ecu.kt
package com.demo.server.ecu

import com.demo.server.common.AbstractJpaPersistable
import com.demo.server.software.Software
import com.demo.server.vehicle.Vehicle
import java.math.BigInteger
import javax.persistence.Entity
import javax.persistence.ManyToMany
import javax.persistence.Table

@Entity
@Table(name = "ecu")
class Ecu(@field:ManyToMany(mappedBy = "supportedEcus")
          var software: List<Software>,
          @field:ManyToMany(mappedBy = "ecus")
          var vehicles: List<Vehicle>) : AbstractJpaPersistable<BigInteger>()